src/rust/bitbox02-noise/src/handshake.rs

```diff
@@ -0,0 +1,55 @@
+// Copyright 2020 Shift Cryptosecurity AG
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! This module implements the X25519 trait needed by noise_protocol
+//! by using the x25519_dalek crate. It is adapted from
+//! https://github.com/sopium/noise-rust/blob/2e4bad0dbe3c674d4e1820e2325ad108c1150dfb/noise-rust-crypto/src/lib.rs#L29,
```

Comment:
need to update this to their new version: https://github.com/sopium/noise-rust/blob/76fb694f06b429879c264087f496958a99710356/noise-rust-crypto/src/lib.rs#L31